### PR TITLE
overlord/devicestate: backoff between retries if the server seems to have refused the serial-request

### PR DESCRIPTION
**Pull request by pedronis**
Also do a better effort extracting the server error message.

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -113,6 +116,21 @@ func (m *DeviceManager) changeInFlight(kind string) bool {
 	return false
 }
 
+func (m *DeviceManager) ensureOperationalBackoff(now time.Time) bool {
```
Comment (zyga) [inline overlord/devicestate/devicemgr.go]: It would be nice to document the meaning of the return value.

File: overlord/devicestate/devicemgr.go [hunk 2]
```diff
@@ -142,11 +160,6 @@ func (m *DeviceManager) ensureOperational() error {
 		return nil
 	}
 
-	if serialRequestURL == "" {
```
Comment (zyga) [inline overlord/devicestate/devicemgr.go]: Why did you remove it? Is it always non-null?

File: overlord/devicestate/devicemgr.go (hunk 1 above)
Comment (zyga) [inline overlord/devicestate/devicemgr.go]: Or maybe rename the function to `shouldEnsureOperationalBackoff`

File: overlord/devicestate/devicemgr.go (hunk 2 above)
Comment (pedronis) [inline overlord/devicestate/devicemgr.go]: @zyga it's a top level var,  this check is a left over from when the feature was wip

Comment (zyga) [inline overlord/devicestate/devicemgr.go]: Ah, thanks for explaining this.

File: overlord/devicestate/devicemgr.go [hunk 3]
```diff
@@ -113,6 +116,21 @@ func (m *DeviceManager) changeInFlight(kind string) bool {
...
 
+func (m *DeviceManager) ensureOperationalBackoff(now time.Time) bool {
+	if !m.lastBecomeOperationalAttempt.IsZero() && m.lastBecomeOperationalAttempt.Add(m.becomeOperationalBackoff).After(now) {
+		return true
+	}
+	if m.becomeOperationalBackoff == 0 {
+		m.becomeOperationalBackoff = 5 * time.Minute
+	} else {
+		if m.becomeOperationalBackoff < (160 * time.Minute) {
+			m.becomeOperationalBackoff *= 2
```
Comment (chipaca) [inline overlord/devicestate/devicemgr.go]: wouldn't it make sense to cap the backoff at, say, a day?
